skbio/embedding/_embedding.py

```diff
@@ -98,3 +108,158 @@ def __repr__(self):
             f"embedding dimension: {dim}\n{indent}has gaps",
         )
         return rstr
+
+
+class SequenceVector(Embedding):
+    r"""Store a vector representation for a biological sequence."""
+    def __init__(self, vector, sequence, **kwargs):
+
+        if isinstance(sequence, Sequence):
+            sequence = str(sequence)
+        if isinstance(sequence, str):
+            sequence = sequence.encode("ascii")
+
+        if len(vector.shape) == 1:
+            vector = vector.reshape(1, -1)
+        if len(vector.shape) == 2:
+            assert vector.shape[0] == 1, (
+                "Only 1 vector per sequence is allowed."
+            )
+
+        seq = np.array([sequence], dtype='O')
+        super(SequenceVector, self).__init__(vector, seq, **kwargs)
+
+    def __str__(self):
+        return str(self._ids[0].decode('ascii'))
+
+    @property
+    def sequence(self):
+        r""" String representation of the underlying sequence """
+        return str(self)
+
+    @property
+    def vector(self):
+        return self.embedding.squeeze()
+
+    def __repr__(self):
+        r"""
+        Return a string representation of the SequenceEmbedding object.
+
+        Returns
+        -------
+        str
+            A string representation of the SequenceEmbedding object.
+
+        See Also
+        --------
+        Protein
+        """
+        seq = Sequence(str(self))
+
+        rstr = repr(seq)
+        rstr = rstr.replace("Sequence", "SequenceVector")
+        n_indent = 4  # see Sequence.__repr__
+        indent = " " * n_indent
+        dim = self.embedding.shape[1]
+        rstr = rstr.replace(
+            "has gaps",
+            f"embedding dimension: {dim}\n{indent}has gaps",
+        )
+        return rstr
+
+    @property
+    def embedding(self):
+        r""" The embedding tensor. """
+        return self._embedding.reshape(1, -1)
+
+    @staticmethod
+    def to_numpy(sequence_vectors : List["SequenceVector"]):
```

Comment:
I don't think we use type annotations elsewhere. If this is kept, shouldn't it be `List[SequenceVector]`?

Reply:
I'm OK with removing type annotations and moving it to the docs

Regarding `List[SequenceVector]`, that won't here, since the `SequenceVector` object needs to be first created before evaluating the `SequenceVector` type

Reply:
Thanks! 

🐔 🥚